File: blue-logistic/src/main/java/com/bluelogistic/entity/Package.java
package com.bluelogistic.entity;

import com.bluelogistic.entity.enums.PackageStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "packages")
@Getter
@Setter
@NoArgsConstructor
public class Package {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private Seller seller;

    @Column(name = "tracking_number", unique = true)
    private String trackingNumber;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal weight;

    @Column(name = "customer_name", nullable = false)
    private String customerName;

    @Column(name = "customer_email", nullable = false)
    private String customerEmail;

    @Column(name = "customer_phone", nullable = false)
    private String customerPhone;

    @Column(name = "delivery_address", nullable = false, columnDefinition = "TEXT")
    private String deliveryAddress;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PackageStatus status = PackageStatus.CREATED;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: blue-logistic-frontend/src/types/package.ts
export type PackageStatus = "CREATED" | "IN_STORAGE" | "DISPATCHED";

export interface Package {
  id: string;
  sellerId: string;
  sellerName: string;
  trackingNumber: string | null;
  description: string;
  weight: number;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  deliveryAddress: string;
  status: PackageStatus;
  createdAt: string;
  updatedAt: string;
  destinationCountry: string;
  costPrice?: number;      // only present for admin
  sellerPrice: number;
  priceBreakdown?: string; // only present for admin
}

export interface CreatePackageRequest {
  description: string;
  weight: number;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  deliveryAddress: string;
  destinationCountry: string;
}

export interface UpdateStatusRequest {
  status: PackageStatus;
}

export interface UpdateTrackingRequest {
  trackingNumber: string;
}
